joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -505,6 +514,8 @@ CallbackReturn JointTrajectoryController::on_configure(const rclcpp_lifecycle::S
     contains_interface_type(command_interface_types_, hardware_interface::HW_IF_VELOCITY);
   has_acceleration_command_interface_ =
     contains_interface_type(command_interface_types_, hardware_interface::HW_IF_ACCELERATION);
+  has_effort_command_interface_ =
+    contains_interface_type(command_interface_types_, hardware_interface::HW_IF_EFFORT);
 
   if (has_velocity_command_interface_)
```

Comment:
It would be good to have also checks if command interfaces are set properly. I guess that we are currently supporting "effort" as use-only command interface.

Can you add those after line 544, similarly as done here, but without “else”. Important is to give error output if “effort” and one other command interface are defined.

Also, according to tests, it appears that the effort interface has as a requirement `"position"` and  `"velocity"` state interfaces. Can you check this also? This would help users to find faster issues with illegal configurations.

(It seems that this last check is missing for the velocity-only interface too...)